frontend/src/libs/hooks/use-search/use-search.hook.ts

```diff
@@ -0,0 +1,23 @@
+import { useMemo, useState } from '../hooks.js';
+
+const useSearch = <T>(
+  elements: T[],
+  propertyName: keyof T,
+): {
+  filteredElements: T[];
+  setFilter: React.Dispatch<React.SetStateAction<string>>;
+} => {
```

Comment:
Create a separate type ReturnType for this

Reply:
done